### PR TITLE
Shyft/rollback

### PR DESCRIPTION
**Pull request by dutterbutter**
**Changes made in this pull request:**

- Added func GetBlockHashesSinceLastValidBlockHash that takes in a common.Hash (block hash) and returns a slice of all the block hashes from that point forward. 

For example, if I passed in the block hash for block 10, acting as the last valid block and the current block height is at 20. The function would return a slice of all block hashs from blocks 11-20. 

- Added func ShyftRollback that takes in a slice of common.Hashs and sets the canonical head to the last valid block height based on the slice passed in. 

- Added a unit test, in order to test the above functionality. The test exists in blockchain_test.go under func TestGetBlockHashesSinceLastValidBlockHash 
- You can run this test by
      - cd core
      - go test -run TestGetBlockHashesSinceLastValidBlockHash

### CODE REVIEW

**Comment by GregTheGreek** [comment]
@dutterbutter you've got lots of conflicts

File: accounts/abi/bind/backends/simulated.go [hunk 1]
```diff
@@ -47,6 +48,11 @@ var _ bind.ContractBackend = (*SimulatedBackend)(nil)
 var errBlockNumberUnsupported = errors.New("SimulatedBackend cannot access blocks other than the latest block")
 var errGasEstimationFailed = errors.New("gas required exceeds allowance or always failing transaction")
 
+// @SHYFT NOTE: test ShyftTracer
+const (
+	testAddress = "0x8605cdbbdb6d264aa742e77020dcbc58fcdce182"
+)
+
```
Comment (GregTheGreek) [inline accounts/abi/bind/backends/simulated.go:55]: what is this for?

File: accounts/abi/bind/util_test.go [hunk 2]
```diff
@@ -51,6 +58,13 @@ var waitDeployedTests = map[string]struct {
 	},
 }
 
+// @SHYFT NOTE: Setup DB for Testing Before Each Test
+// func TestMain(m *testing.M) {
+// 	shyfttest.PgTestDbSetup()
+// 	retCode := m.Run()
+// 	shyfttest.PgTestTearDown()
+// 	os.Exit(retCode)
+// }
```
Comment (GregTheGreek) [inline accounts/abi/bind/util_test.go]: can we remove that?

**Review by GregTheGreek — APPROVED** [review]

**Comment by GregTheGreek** [comment]
@dbkbali can you have a look. Seems to be good on my end.

**Comment by GregTheGreek** [comment]
@dbkbali Is this good to go?

**Review by dbkbali — APPROVED** [review]
I am would normally be concerned about merging all these test fixes into development branch currently as it includes much of the test refactoring that we worked on previously. All of these test fixes - for the old geth tests will likely need refactoring due to changes that will be introduced when we merge the ring_node sprint work into development due to pg db changes in the current rollback work. In the future we should endeavour to keep merge request to cover working features only with all tests green. Given that we are aware that in the existing development branch the original geth tests still have failures I dont have a problem approving this - as I am aware our next priority will be to address the failing original geth tests and getting them to green - when that happens we should strictly adhere to merges only being approved when all tests are green and hopefully have our CI server reconfigured to test this on merge requests

**Comment by dutterbutter** [comment]
Been merged into a base branch

**Pull request closed without merging**